x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceLicenseService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicensedFeature;
+import org.elasticsearch.license.XPackLicenseState;
+
+/**
+ * Determines based on license and fallback setting whether synthetic source usages should fallback to stored source.
+ */
+public final class SyntheticSourceLicenseService {
+
+    public static final String LOGSDB_FEATURE_FAMILY = "logsdb";
+
+    /**
+     * A setting that determines whether source mode should always be stored source. Regardless of licence.
+     */
+    public static final Setting<Boolean> FALLBACK_SETTING = Setting.boolSetting(
+        "xpack.logsdb.synthetic_source_fallback_to_stored_source",
```

Comment:
This is not specific to logsdb right?

Reply:
That is true. Maybe `xpack.mapper.synthetic_source_fallback_to_stored_source` is better?